mteb/abstasks/AbsTask.py

```diff
@@ -26,23 +26,27 @@ def __init__(self, seed=42, **kwargs):
         torch.manual_seed(self.seed)
         torch.cuda.manual_seed_all(self.seed)
 
+    def dataset_transform(self):
+        """
+        Transform operations applied to the dataset after loading.
+        Override this method if your dataset requires any transformation.
+        """
+        pass
+
     def load_data(self, **kwargs):
         """
         Load dataset from HuggingFace hub
         """
         if self.data_loaded:
             return
-
-        # TODO: add split argument
-        self.dataset = datasets.load_dataset(
-            self.metadata_dict["hf_hub_name"],
-            revision=self.metadata_dict.get("revision", None),
-        )
+        self.dataset = datasets.load_dataset(**self.metadata_dict["dataset"])
```

Comment:
```suggestion
        self.dataset = datasets.load_dataset(
             self.metadata. hf_hub_name,
             revision=self.metadata.revision,
             **self.metadata.dataset_kwargs
         )
```

thanks @gariepyalex this seems like a very solid commit. I would generally love that almost always specify the revision (unless we have good reason not to), I am afraid that the current structure would lead to us not specifying the revision. One option might be to allow a dataset_kwargs, which is not a the revision or HF name.

Let me know what you think about this approach.

Reply:
Fair feedback. What I like about the proposed solution is all the arguments passed to `load_dataset` are grouped together. `hf_hub_name` is a name that only exist in MTEB, and it is not immediately obvious that it refers to the path argument of `load_dataset`. 

> I am afraid that the current structure would lead to us not specifying the revision.

An alternative would be to add an additional validator on the `dataset` field of the metadata to ensure that the revision is specified:

```python
@field_validator("dataset")
    def _check_dataset_revision_is_specified(cls, dataset):
        if "revision" not in dataset:
            raise ValueError("You must explicitly specify a revision for the dataset (either a SHA or None).")
        if dataset["revision"] is None:
            logging.warning("It is encourage to specify a dataset revision for reproducability")
        return dataset
```

Let me know what you think! I can also implement the `dataset_kwargs` solution you proposed if you prefer.

Reply:
I can see good reason in that. It would also allow us to change out the dataset source reasonably well in the future. 

I would add a validator for dataset["path"] as well.
I would also rename the field to `load_dataset_kwargs` or `datasets_kwargs` to make it clear what it is being passed to.

Reply:
@KennethEnevoldsen 
I understand your point when saying that renaming the `dataset` field to `load_dataset_kwargs` or `datasets_kwargs`. However, to me, it feels a bit strange to name the field that way. Everything seems to have a declarative format in the TaskMetadata object. The kwargs part feels a bit more functional nomenclature.
That being said, I won't die on a hill for that. Let us know if you still want the change.

Reply:
Keep it as `dataset`

Reply:
All validators were added